packages/web3-eth/src/rpc_methods.ts

```diff
@@ -0,0 +1,518 @@
+) {
+	validateHexString32Bytes(blockHash);
+
+	return requestManager.send<'eth_getUncleCountByBlockHash'>({
+		method: 'eth_getUncleCountByBlockHash',
+		params: [blockHash],
+	});
+}
+
+export async function getUncleCountByBlockNumber(
+	requestManager: Web3RequestManager,
+	blockNumber: BlockNumberOrTag,
+) {
+	validateBlockNumberOrTag(blockNumber);
+
+	return requestManager.send<'eth_getUncleCountByBlockNumber'>({
+		method: 'eth_getUncleCountByBlockNumber',
+		params: [blockNumber],
+	});
+}
+
+export async function getCode(
+	requestManager: Web3RequestManager,
+	address: Address,
+	blockNumber: BlockNumberOrTag,
+) {
+	validateAddress(address);
+	validateBlockNumberOrTag(blockNumber);
+
+	return requestManager.send<'eth_getCode'>({
+		method: 'eth_getCode',
+		params: [address, blockNumber],
+	});
+}
+
+export async function sign(
+	requestManager: Web3RequestManager,
+	address: Address,
+	message: HexStringBytes,
+) {
+	validateAddress(address);
+	validateHexStringInput(message);
+
+	return requestManager.send<'eth_sign'>({
+		method: 'eth_sign',
+		params: [address, message],
+	});
+}
+
+export async function signTransaction(
+	requestManager: Web3RequestManager,
+	transaction: TransactionWithSender,
+) {
+	validateTransactionWithSender(transaction);
+
+	return requestManager.send<'eth_signTransaction'>({
+		method: 'eth_signTransaction',
+		params: [transaction],
+	});
+}
+
+export async function sendTransaction(
+	requestManager: Web3RequestManager,
+	transaction: TransactionWithSender,
+) {
+	validateTransactionWithSender(transaction);
+
+	return requestManager.send<'eth_sendTransaction'>({
+		method: 'eth_sendTransaction',
+		params: [transaction],
+	});
+}
+
+export async function sendRawTransaction(
+	requestManager: Web3RequestManager,
+	transaction: HexStringBytes,
+) {
+	validateHexStringInput(transaction);
+
+	return requestManager.send<'eth_sendRawTransaction'>({
+		method: 'eth_sendRawTransaction',
+		params: [transaction],
+	});
+}
+
+export async function call(
+	requestManager: Web3RequestManager,
+	transaction: TransactionCall,
+	blockNumber: BlockNumberOrTag,
+) {
+	validateTransactionCall(transaction);
+	validateBlockNumberOrTag(blockNumber);
+
+	return requestManager.send<'eth_call'>({
+		method: 'eth_call',
+		params: [transaction, blockNumber],
+	});
+}
+
+// TODO Not sure how to best validate Partial<TransactionWithSender>
+export async function estimateGas(
+	requestManager: Web3RequestManager,
+	transaction: Partial<TransactionWithSender>,
+	blockNumber: BlockNumberOrTag,
+) {
+	validateBlockNumberOrTag(blockNumber);
+
+	return requestManager.send<'eth_estimateGas'>({
+		method: 'eth_estimateGas',
+		params: [transaction, blockNumber],
+	});
+}
+
+export async function getBlockByHash(
+	requestManager: Web3RequestManager,
+	blockHash: HexString32Bytes,
+	hydrated: boolean,
+) {
+	validateHexString32Bytes(blockHash);
+	validateBoolean(hydrated);
+
+	return requestManager.send<'eth_getBlockByHash'>({
```

Comment:
While you are using the `{method: 'eth_getBlockByHash'}` you don't need to pass the generic value here. It will infer automatically from the parameter.